pkg/identities/service.go

```diff
@@ -163,9 +168,62 @@ func (s *Service) CreateIdentity(ctx context.Context, bodyID *kClient.CreateIden
 
 	s.authz.SetCreateIdentityEntitlements(ctx, identity.Id)
 
+	err = s.SendUserCreationEmail(ctx, identity)
+
 	return data, err
 }
 
+func (s *Service) SendUserCreationEmail(ctx context.Context, identity *kClient.Identity) error {
+	ctx, span := s.tracer.Start(ctx, "identities.Service.SendUserCreationEmail")
+	defer span.End()
+
+	template, err := mail.LoadTemplate(mail.UserCreationInvite)
+	if err != nil {
+		return err
+	}
+
+	code, link, err := s.generateRecoveryInfo(ctx, identity.Id)
+	if err != nil {
+		return err
+	}
+
+	emailAddress := ""
+	if e, ok := identity.Traits.(map[string]interface{})["email"]; ok {
+		emailAddress = e.(string)
+	}
+
+	if emailAddress == "" {
+		return fmt.Errorf("\"email\" address not found in identity traits")
+	}
+
+	userCreationInviteArgs := mail.UserCreationInviteArgs{
+		Email:        emailAddress,
+		InviteUrl:    link,
+		RecoveryCode: code,
+	}
+
+	err = s.email.Send(ctx, emailAddress, userCreationEmailSubject, template, userCreationInviteArgs)
```

Comment:
If this fails the identity will be created, but we will return an error. This means that the user will be unable to input the same username when the error is fixed, but they will also be unable to access this account.

Reply:
Yes, in this case the admin will have to remove the identity and create it again.
We can improve on this in the future